tldw_Server_API/app/core/Evaluations/recipes/rag_retrieval_tuning_candidates.py

```
@@ -0,0 +1,328 @@
+        raise ValueError("top_k must be a positive integer.")
+    if normalized["rerank_top_k"] <= 0:
+        raise ValueError("rerank_top_k must be a positive integer.")
+    return normalized
+
+
+def _normalize_indexing_config(indexing_config: Mapping[str, Any]) -> dict[str, Any]:
+    if not indexing_config:
+        return {}
+    unknown_keys = sorted(set(indexing_config) - _INDEXING_KNOBS)
+    if unknown_keys:
+        raise ValueError(f"unsupported candidate knob: {unknown_keys[0]}")
```

Ruff (0.15.7)

[warning] Avoid specifying long messages outside the exception class

(TRY003)